### PR TITLE
MNT: Minor update to PR template

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address the following comments on the PR template:

* https://github.com/astropy/astropy/pull/11844#issuecomment-865075432
* https://github.com/astropy/astropy/pull/11864#issuecomment-865189347

This is a direct follow up of #11844 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will
review this pull request of some common things to look for. This list
is not exhaustive.

Primary reviewer GitHub name: @taldcroft

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] If the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -51,9 +51,7 @@ Fixes #<Issue Number>
 <!-- This section is to be filled by package maintainer(s) who will
 review this pull request. -->
 
-This checklist is meant to remind the package maintainer(s) who will
-review this pull request of some common things to look for. This list
-is not exhaustive.
+This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.
```
Comment (taldcroft) [inline .github/PULL_REQUEST_TEMPLATE.md:54]: What about a line like below?
```
Primary reviewer GitHub name: [edit and fill in]
```

**Review by taldcroft — APPROVED** [review]
I'm OK with this as-is, but would also be OK with the change I suggested.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (bsipocz-2nd) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: I'm somewhat :-1 on this, after all the definition of subpackage maintainers are exactly that they do the review and sign-off on PRs. 
(I find editing descriptions and checklist taking unnecessary time for those doing the workflow without reminder, and working mostly with quick, trivial PRs. 
So, I would aim for the middle ground, keep the list short and focused that ensure those who didn't do these checked on their own would do it. Asking them to fill out even more boxes and edits I feel is even more counter-productive).

Comment (bsipocz) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: I'm somewhat :-1 on this, after all the definition of subpackage maintainers are exactly that they do the review and sign-off on PRs.
(I find editing descriptions and checklist taking unnecessary time for those doing the workflow without reminder, and working mostly with quick, trivial PRs.
So, I would aim for the middle ground, keep the list short and focused that ensure those who didn't do these checked on their own would do it. Asking them to fill out even more boxes and edits I feel is even more counter-productive).

Comment (pllim) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: How about this, I'll let leave this PR open for a few more days to collect opinions. If I get no more opinions in support of this request, I'll _not_ include "primary reviewer" field.

In a way, you can already see who checked what via the GitHub issue edit history, though it takes time to dig through the dropdown menu.

Screenshot 2021-06-28 160352

Comment (mhvk) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: I'm with @bsipocz here - let's not require editing of the entry - between switching to the review page, then to the bottom of conversation page for CI results and latest discussion, back to the top of conversation page for checklist, and then finally again to the bottom of the page for merging, it is already a bit of a hassle.

Comment (taldcroft) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: That's fine to skip the primary review edit. 

It's worth noting however that the reviewer should already be looking at the top of the conversation carefully to check the milestone and labels, and the bottom to make sure CI is OK. So clicking the boxes does not take substantial extra time. Even on trivial ones if a reviewer is too busy for that 15 seconds then I wonder if the review was thorough. (For me, the median time I spend on reviewing PRs is probably more than 15 *minutes*, so 15 seconds at the end is truly in the noise).

Comment (mhvk) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: Fair enough. I certainly have done a few reviews too quickly and had decided independently that it was better to somewhat reduce the number of PRs I reviews (hopefully increasing the quality...).

Comment (embray) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: I don't understand what this is changing.  It just looks like it unwrapped this line?  I don't know why it's garnering additional discussion--or is it just that people didn't take a close look at this checklist on the original PR?

Comment (pllim) [reply .github/PULL_REQUEST_TEMPLATE.md:54]: @embray , the discussion here is about https://github.com/astropy/astropy/pull/11870/files#r660017272 .

**Comment by pllim** [comment]
Okay, I'll wait till Friday (2021-07-02) in case there are more opinions. But feel free to merge earlier if you think the discussions are done. Thanks, everyone!

**Comment by pllim** [comment]
I don't see more comments, so I merged this. Thanks again!

**Pull request merged**